decidim-blogs/spec/system/explore_posts_spec.rb

```diff
@@ -50,6 +50,18 @@
         expect(page).to have_css("[data-pages] [data-page]", count: 2)
       end
     end
+
+    context "with some unpublished posts" do
```

Comment:
Please also add a spec that if the blog post is unpublished, the user cannot access the blog post's page and should receive a HTTP 404 instead.